packages/console/src/monaco/MonacoProviders.tsx

```diff
@@ -62,17 +83,12 @@ class MonacoProviders extends PureComponent<
   ): Promise<void> {
     MonacoProviders.ruffSettings = settings;
 
-    // Ruff has not been initialized yet
-    if (
-      MonacoProviders.ruffWorkspace == null &&
-      MonacoProviders.isRuffEnabled
-    ) {
+    if (!MonacoProviders.isRuffInitialized) {
       MonacoProviders.initRuff();
```

Comment:
Should we be awaiting this Promise? Otherwise not suer why setRuffSettings is marked as `async`

Reply:
Nope doesn't look like we await it where it's called either (confirmed in enterprise as well).

Probably originally I wrote it with something async and just didn't catch that it was unnecessary in a refactor. Looks like there's an eslint rule if we wanted it [here](https://typescript-eslint.io/rules/require-await/)

Instead this now initializes ruff if needed which will apply the linter after initialization finishes, or if already initialized update the workspace which will do the linting. Awaiting the promise and removing the return would run the linter twice in quick succession.